https://redhat.atlassian.net/browse/ACM-30036 (#8746)
open-cluster-management/rhacm-docs: gitops/gitops_configure.adoc

:_mod-docs-content-type: PROCEDURE

[#configure-subs-resources-gitops]
= Configuring subscriptions and resources for Argo CD 

[role="_abstract"]
To enable the `ArgoCD` agent, you must configure the {gitops-short} operator subscription and the `ArgoCD` custom resource. To configure the necessary subscriptions and resources, complete the following steps:

*Required access*: Cluster administrator

.Procedure

> . On the hub cluster only, modify the {gitops-short} operator subscription to include the required environment variables by running the following command:

Reviewer: ```suggestion
. On the hub cluster only, run the following command to modify the {gitops-short} operator subscription to include the required environment variables:

+
[source,bash]
----
oc edit subscription gitops-operator -n openshift-gitops-operator
----

> . Add the following environment variables to the `spec.config.env` file by adding the following YAML sample:

Reviewer: ```suggestion
. Add the following environment variables to the `spec.config.env` file. See the following YAML sample `name` and `value` settings:
```

Reviewer: @dockerymick did you not want this one or just didn't see it.

Author: I missed it, thank you!

+
[source,yaml]
----
spec:
  config:
    env:
    - name: ARGOCD_CLUSTER_CONFIG_NAMESPACES
      value: openshift-gitops
    - name: ARGOCD_PRINCIPAL_TLS_SERVER_ALLOW_GENERATE
      value: "false"
    - name: ARGOCD_PRINCIPAL_REDIS_SERVER_ADDRESS
      value: openshift-gitops-redis:6379
----

. Replace the existing `ArgoCD` custom resource with the compatible `Agent` mode configuration by adding the following YAML sample:

> 

Reviewer: ` argoCDAgent:`
